### ISSUE TITLE
Project proposal - techonologies and implementation details

### ISSUE THREAD

**Issue by dakaizou** [umbcs410-utimaco/crypto-analysis#19]
Please describe your choices of technology (if applicable, please distinguish between frontend and backend).

Please model key workflows as UML Sequence Diagrams.

**Comment by dakaizou** [comment]
https://www.cylynx.io/blog/a-comparison-of-javascript-graph-network-visualisation-libraries/
https://ialab.it.monash.edu/webcola/
https://cprimozic.net/blog/speeding-up-webcola-with-webassembly/

**Comment by dakaizou** [comment]
- nodejs
- typescript
- sigmajs
- graphology
- https://www.blockchain.com/api/blockchain_api
- react
- html
- css
- http
- ajax
- github pages
- github action

**Comment by dakaizou** [comment]
```plantuml
@startuml
actor User
participant "Landing Page"
participant Graph
collections "Tx Store"
database "Local Storage"
participant blockchain.com

User -> "Landing Page": request
"Landing Page" -> Graph: request
Graph -> "Tx Store": get labeled addresses
"Tx Store" -> "Local Storage": get labeled addresses
"Tx Store" <- "Local Storage": return labeled addresses
"Tx Store" -> "blockchain.com": get transactions
"Tx Store" -> "blockchain.com": return transactions
Graph <- "Tx Store": return labeled addresses
"Landing Page" <- Graph: draw
User <- "Landing Page": show network graph
@enduml
```

**Comment by dakaizou** [comment]
![sequence](https://user-images.githubusercontent.com/13910784/155914199-e25ab4fd-a47d-4815-a16e-dd0f2b7136e6.svg)

**Comment by dakaizou** [comment]
This project needs to provide a starting point for analytics.
In order to understand the data better, data visualization is one of the most important tool to help us discover information from them. The web application client side technology is one of the most mature tech stack to visualize data and create interactive feature-rich graphical user interface. This characteristic fits well with the other requirement of this project "being able to label and monitor suspicious wallet addresses". Further, browsers are maintained by a lot of people, and they are an unparalleled compatibility layer for cross platform application.

Thus, we decided to create an [single-page application (SPA)](https://en.wikipedia.org/wiki/Single-page_application), since most of the requirements of this project can be fulfilled without a dedicated web server backend. If necessary, we can utilize the [local storage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) for persistent data like labeled addresses.

Moreover, we want to visualize transactions and wallet addresses as a graph, so it is intuitive to operate. [`sigma.js`](https://www.sigmajs.org/) makes working on graphs a lot eaiser in JavaScript/TypeScript, so we went with the `sigma.js` eco system(Node.js, React, sigma.js, graphology, html, css).
